models/jobs.js

```diff
@@ -56,11 +56,15 @@ exports.unlock = function(db, jobId) {
 };
 
 exports.obtainLock = function(db, jobId, callback) {
-  db.query(sqlQueries.obtainLockForJob, [jobId], gotResult);
+  db.query(sqlQueries.obtainLockForJob, [jobId], gotLock);
 
-  function gotResult(err, result) {
+  function gotLock(err, result) {
     if (err) return callback(err);
-    callback(null, result.rows[0]);
+    db.query(sqlQueries.getJobData, [jobId], gotData);
+    function gotData(err, result) {
```

Comment:
this function doesn't really need to be nested like this does it? move it back to the same level as gotLock